examples/bench/bench_results.md

```diff
@@ -4,6 +4,68 @@
 
 Results are total time for 1, 2, 4 threads, on my MacBook Pro (16-inch 2021, Apple M1 Max, max config)
 
+# Axon 1.7.3x (ror/gpu2 branch): GPU first results
+
+This are very first results with no validation of the computation -- main thing is that it doesn't crash and the computational times look plausible!  SynCa is a beast and needs to be investigated!  Using ./run_gpu.sh to target HUGE and LARGE:
+
+HUGE:
```

Comment:
I can't really reproduce these results locally. This is what I'm getting when I run `./run_gpu.sh` (The Mac CPU runs fast at default thread settings).:
```
==============================================================
HUGE Network (5 x 1024 units)
OS Threads (=GOMAXPROCS): 10. Gorountines: 10 (Neurons) 10 (SendSpike) 4 (SynCa)
Total Secs:      6
goos: darwin
goarch: arm64
pkg: github.com/emer/axon/examples/bench
BenchmarkBenchNetFull-10    	       1	7292727208 ns/op
PASS
Running bench with: 0 neuronThreads, 0 sendSpikeThreads, 0 synCaThreads, 2 epochs, 5 pats, 1024 units
Threading: using default values
2023/02/14 11:29:14          Input:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden1:	 Syns: 1048576	 SynnMem: 64.0 MB
       Hidden1:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden2:	 Syns: 1048576	 SynnMem: 64.0 MB
       Hidden2:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden1:	 Syns: 1048576	 SynnMem: 64.0 MB
	       Hidden3:	 Syns: 1048576	 SynnMem: 64.0 MB
       Hidden3:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden2:	 Syns: 1048576	 SynnMem: 64.0 MB
	        Output:	 Syns: 1048576	 SynnMem: 64.0 MB
        Output:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden3:	 Syns: 1048576	 SynnMem: 64.0 MB


      BenchNet:	 Neurons: 5120	 NeurMem: 1.8 MB 	 Syns: 7340032 	 SynMem: 532.0 MB

epc: 0  	CorSim: -0.9999863 	AvgCorSim: -0.99999046 	Time:5.477675s
epc: 1  	CorSim: -0.9999157 	AvgCorSim: -0.9999752 	Time:5.648138s
Took  11.13 secs for 2 epochs, avg per epc:  5.563
TimerReport: BenchNet
OS Threads (=GOMAXPROCS): 10. Gorountines: 10 (Neurons) 10 (SendSpike) 4 (SynCa)
	Function Name 	   Secs	    Pct
	GPU:ApplyExts 	  0.004	    0.0
	GPU:BetweenGi 	  0.437	    4.3
	    GPU:Cycle 	  0.554	    5.4
	GPU:CyclePost 	  0.469	    4.6
	      GPU:DWt 	  0.057	    0.6
	GPU:GatherSpikes 	  0.669	    6.5
	    GPU:LayGi 	  0.458	    4.5
	GPU:MinusPhase 	  0.004	    0.0
	 GPU:NewState 	  0.019	    0.2
	GPU:PlusPhase 	  0.004	    0.0
	   GPU:PoolGi 	  0.439	    4.3
	GPU:SendSpike 	  1.840	   17.9
	GPU:SynCaRecv 	  2.768	   26.9
	GPU:SynCaSend 	  2.500	   24.3
	  GPU:WtFmDWt 	  0.057	    0.6
	        Total 	 10.278
goos: darwin
goarch: arm64
pkg: github.com/emer/axon/examples/bench
BenchmarkBenchNetFull-10    	       1	12429779084 ns/op
--- BENCH: BenchmarkBenchNetFull-10
    bench_test.go:75: skipping convergence test because numEpochs < 350
PASS
Running bench with: 0 neuronThreads, 0 sendSpikeThreads, 0 synCaThreads, 2 epochs, 5 pats, 1024 units
Threading: using default values
2023/02/14 11:29:27          Input:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden1:	 Syns: 1048576	 SynnMem: 64.0 MB
       Hidden1:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden2:	 Syns: 1048576	 SynnMem: 64.0 MB
       Hidden2:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden1:	 Syns: 1048576	 SynnMem: 64.0 MB
	       Hidden3:	 Syns: 1048576	 SynnMem: 64.0 MB
       Hidden3:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden2:	 Syns: 1048576	 SynnMem: 64.0 MB
	        Output:	 Syns: 1048576	 SynnMem: 64.0 MB
        Output:	 Neurons: 1024	 NeurMem: 368.0 KB 	 Sends To:
	       Hidden3:	 Syns: 1048576	 SynnMem: 64.0 MB


      BenchNet:	 Neurons: 5120	 NeurMem: 1.8 MB 	 Syns: 7340032 	 SynMem: 532.0 MB

epc: 0  	CorSim: 0.0068378774 	AvgCorSim: 0.0034149922 	Time:4.684524917s
epc: 1  	CorSim: 0.16637047 	AvgCorSim: 0.036158577 	Time:4.53522025s
Took   9.22 secs for 2 epochs, avg per epc:   4.61
TimerReport: BenchNet
OS Threads (=GOMAXPROCS): 10. Gorountines: 10 (Neurons) 10 (SendSpike) 4 (SynCa)
	Function Name 	   Secs	    Pct
	  CycleNeuron 	  0.415	    4.5
	    CyclePost 	  0.000	    0.0
	          DWt 	  1.672	   18.2
	   DWtSubMean 	  0.000	    0.0
	 GatherSpikes 	  0.161	    1.8
	   GiFmSpikes 	  0.116	    1.3
	PoolGiFmSpikes 	  0.001	    0.0
	    SendSpike 	  1.695	   18.4
	    SynCaRecv 	  0.929	   10.1
	    SynCaSend 	  3.794	   41.2
	      WtFmDWt 	  0.426	    4.6
	        Total 	  9.209
goos: darwin
goarch: arm64
pkg: github.com/emer/axon/examples/bench
BenchmarkBenchNetFull-10    	       1	10365420750 ns/op
--- BENCH: BenchmarkBenchNetFull-10
    bench_test.go:75: skipping convergence test because numEpochs < 350
PASS
```

Particularly the 2.2s GPU number for HUGE is very different from what I'm getting. My CPU number is faster than yours because I'm using more threads I think.

Reply:
It also seems like there isn't much scaling happening, CPU is 10% faster on my machine for both HUGE (CPU 9s, GPU 10s) and GINORMOUS (CPU 16s, GPU 15s). Would've expected the GPU to get faster as the networks get bigger.